extensions/wikia/Recirculation/services/FandomDataService.class.php

```diff
@@ -46,6 +51,66 @@ function() {
 		return $data;
 	}
 
+	private function setupCategories( $ignoreTopic ) {
```

Comment:
This function changes the internal state of the object ($this->option['categories']) but isn't set in the constructor, but rather in getPosts. This means a getter changes the state of the object. It seems wrong.